libcalico-go/lib/backend/watchersyncer/watchersyncer.go

```diff
@@ -60,29 +74,44 @@ type SyncerUpdateProcessor interface {
 	OnSyncerStarting()
 }
 
+type Option func(*watcherSyncer)
+
+func WithWatchRetryTimeout(t time.Duration) Option {
+	return func(ws *watcherSyncer) {
+		ws.watchRetryTimeout = t
+	}
+}
+
+var _ = WithWatchRetryTimeout
```

Comment:
```suggestion
var _ Option = WithWatchRetryTimeout
```

Reply:
I think this is what you're going for?

Reply:
I was just trying to stop it from being completely unused, but agree that's better since it checks the type too.

Reply:
Actually that doesn't type check, would need to call the func :-(